pkg/network/setup/configstate.go

```diff
@@ -42,7 +42,7 @@ func NewConfigState(configStateCache configStateCacheInterface) ConfigState {
 //
 // The discovery step can be executed repeatedly with no limitation.
 // The configuration step is allowed to run only once. Any attempt to run it again will cause a critical error.
-func (c ConfigState) Run(nics []podNIC, discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
+func (c ConfigState) Run(nics []podNIC, discoverFunc func(*podNIC) error, configFunc func(*podNIC) error, ns NSExecutor) error {
```

Comment:
Question: Why it was not added to the ConfigState? I think  the input is available when the object is created, no?